tests/eosio.wrap_tests.cpp

```diff
@@ -333,35 +340,52 @@ BOOST_FIXTURE_TEST_CASE( wrap_with_msig_producers_change, eosio_wrap_tester ) tr
-
-   BOOST_REQUIRE_EQUAL( 1, traces[1]->action_traces.size() );
-   BOOST_REQUIRE_EQUAL( config::system_account_name, name{traces[1]->action_traces[0].act.account} );
-   BOOST_REQUIRE_EQUAL( "reqauth"_n, name{traces[1]->action_traces[0].act.name} );
-   BOOST_REQUIRE_EQUAL( transaction_receipt::executed, traces[1]->receipt->status );
+   BOOST_REQUIRE( bool(trx_trace) );
+   BOOST_REQUIRE_EQUAL( transaction_receipt::executed, trx_trace->receipt->status );
+   BOOST_REQUIRE_EQUAL( 3, trx_trace->action_traces.size() );
+
+   auto msig_exec_trace = trx_trace->action_traces.at(0);
+   BOOST_REQUIRE_EQUAL( 1, msig_exec_trace.action_ordinal );
+   BOOST_REQUIRE_EQUAL( 0, msig_exec_trace.creator_action_ordinal );
+   BOOST_REQUIRE_EQUAL( 0, msig_exec_trace.closest_unnotified_ancestor_action_ordinal );
+   BOOST_REQUIRE_EQUAL( "eosio.msig"_n, msig_exec_trace.receiver );
+   BOOST_REQUIRE_EQUAL( "eosio.msig"_n, msig_exec_trace.act.account );
+   BOOST_REQUIRE_EQUAL( "exec"_n, msig_exec_trace.act.name );
+   BOOST_REQUIRE_EQUAL( 1, msig_exec_trace.act.authorization.size() );
+   BOOST_REQUIRE_EQUAL( "alice"_n, msig_exec_trace.act.authorization.at(0).actor );
+   BOOST_REQUIRE_EQUAL( "active"_n, msig_exec_trace.act.authorization.at(0).permission );
+
+   auto wrap_exec_trace = trx_trace->action_traces.at(1);
+   BOOST_REQUIRE_EQUAL( 2, wrap_exec_trace.action_ordinal );
+   BOOST_REQUIRE_EQUAL( 1, wrap_exec_trace.creator_action_ordinal );
+   BOOST_REQUIRE_EQUAL( 1, wrap_exec_trace.closest_unnotified_ancestor_action_ordinal );
+   BOOST_REQUIRE_EQUAL( "eosio.wrap"_n, wrap_exec_trace.receiver );
+   BOOST_REQUIRE_EQUAL( "eosio.wrap"_n, wrap_exec_trace.act.account );
+   BOOST_REQUIRE_EQUAL( "exec"_n, wrap_exec_trace.act.name );
+   BOOST_REQUIRE_EQUAL( 2, wrap_exec_trace.act.authorization.size() );
+   BOOST_REQUIRE_EQUAL( "alice"_n, wrap_exec_trace.act.authorization.at(0).actor );
+   BOOST_REQUIRE_EQUAL( "active"_n, wrap_exec_trace.act.authorization.at(0).permission );
+   BOOST_REQUIRE_EQUAL( "eosio.wrap"_n, wrap_exec_trace.act.authorization.at(1).actor );
+   BOOST_REQUIRE_EQUAL( "active"_n, wrap_exec_trace.act.authorization.at(1).permission );
+
+   auto reqauth_trace = trx_trace->action_traces.at(2);
+   BOOST_REQUIRE_EQUAL( 3, reqauth_trace.action_ordinal );
+   BOOST_REQUIRE_EQUAL( 2, reqauth_trace.creator_action_ordinal );
+   BOOST_REQUIRE_EQUAL( 2, reqauth_trace.closest_unnotified_ancestor_action_ordinal );
+   BOOST_REQUIRE_EQUAL( config::system_account_name, reqauth_trace.receiver );
+   BOOST_REQUIRE_EQUAL( config::system_account_name, reqauth_trace.act.account );
+   BOOST_REQUIRE_EQUAL( "reqauth"_n, reqauth_trace.act.name );
+   BOOST_REQUIRE_EQUAL( 1, reqauth_trace.act.authorization.size() );
+   BOOST_REQUIRE_EQUAL( "bob"_n, reqauth_trace.act.authorization.at(0).actor );
+   BOOST_REQUIRE_EQUAL( "active"_n, reqauth_trace.act.authorization.at(0).permission );
```

Comment:
This level of detailed checking is okay, but not necessary. Just checking the receiver and action names in the proper order would have been enough.

Reply:
done